dicom-archive/updateMRI_Upload.pl

```diff
@@ -126,64 +145,55 @@
 ################################################################
 
 # fetch tarchiveLibraryDir from ConfigSettings in the database
-my $tarchiveLibraryDir = &NeuroDB::DBI::getConfigSetting(
-                            \$dbh,'tarchiveLibraryDir'
-                            );
+
+my $configOB = NeuroDB::objectBroker::ConfigOB->new(db => $db);
+my $tarchiveLibraryDir = $configOB->getTarchiveLibraryDir();
+
 # determine tarchive path stored in the database (without tarchiveLibraryDir)
 my $tarchive_path = $tarchive;
-$tarchive_path    =~ s/$tarchiveLibraryDir\/?//g;  
+$tarchive_path    =~ s/$tarchiveLibraryDir\/?//g;
 
-my $where         = " WHERE t.ArchiveLocation =?";
-
-if ($globArchiveLocation) {
-    $where = " WHERE t.ArchiveLocation LIKE ?";
-    $tarchive_path = basename($tarchive);
-}
+# Check if there is already an mri upload record for the tarchive
+my $mriUploadOB = NeuroDB::objectBroker::MriUploadOB->new(db => $db);
+my $resultRef = $mriUploadOB->getWithTarchive(
+    GET_COUNT, $tarchive_path, $globArchiveLocation
+);
 
-($query = <<QUERY) =~ s/\n/ /gm;
-SELECT 
-  COUNT(*) 
-FROM 
-  mri_upload m 
-JOIN 
-  tarchive t ON (t.TarchiveID=m.TarchiveID) 
-QUERY
-$query .= $where;
-$sth = $dbh->prepare($query);
-$sth->execute($tarchive_path);
-my $count = $sth->fetchrow_array;
-if($count>0) {
-   print "\n\tERROR: the tarchive is already uploaded \n\n"; 
+if($resultRef->[0]->[0] > 0) {
+   print "\n\tERROR: the tarchive is already uploaded \n\n";
    exit 6;
-} 
-
+}
 
 ################################################################
 #####get the tarchiveid from tarchive table#####################
 ################################################################
-($query = <<QUERY) =~ s/\n/ /gm;
-SELECT 
-  t.TarchiveID 
-FROM 
-  tarchive t 
-QUERY
-$query .= $where;
-$sth = $dbh->prepare($query);
-$sth->execute("%".$tarchive_path."%");
-my $tarchiveID = $sth->fetchrow_array;
 
+my $tarchiveOB = NeuroDB::objectBroker::TarchiveOB->new(db => $db);
+$resultRef = $tarchiveOB->getByTarchiveLocation(
+    ['TarchiveID'], $tarchive_path, $globArchiveLocation
+);
+
+if(@$resultRef != 1) {
+    die sprintf(
+        "Unexpected number of tarchive records with location %s found: %d\n",
+        $tarchive_path,
+        scalar(@$resultRef)
+    );
+}
+my $tarchiveID = $resultRef->[0]->[0];
```

Comment:
will these eventually be the column name instead of the index?

Reply:
Yes